### PR TITLE
[Form] Add bool type to ChoiceListFactoryInterface::createView $label argument

### PR DESCRIPTION
**Pull request by vudaltsov**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | n/a
| License       | MIT
| Doc PR        | n/a

Replaces #33074 
Ping @nicolas-grekas , @Tobion , @yceruto .

### CODE REVIEW

**Review by yceruto — APPROVED** [review]
The next step would be to add the \TypeError in master/5.0, no?

File: src/Symfony/Component/Form/ChoiceList/Factory/ChoiceListFactoryInterface.php [hunk 1]
```diff
@@ -78,7 +78,7 @@ public function createListFromLoader(ChoiceLoaderInterface $loader, $value = nul
      * attributes that should be added to the respective choice.
      *
      * @param array|callable|null $preferredChoices The preferred choices
-     * @param callable|null       $label            The callable generating the
+     * @param callable|bool|null  $label            The callable generating the
      *                                              choice labels
```
Comment (yceruto) [inline src/Symfony/Component/Form/ChoiceList/Factory/ChoiceListFactoryInterface.php]: btw, we should explain the `false` value here.

**Comment by vudaltsov** [comment]
> The next step would be to add the \TypeError in master/5.0, no?

Yes, I will that it in #32237

File: src/Symfony/Component/Form/ChoiceList/Factory/ChoiceListFactoryInterface.php (hunk 1 above)
Comment (vudaltsov) [reply src/Symfony/Component/Form/ChoiceList/Factory/ChoiceListFactoryInterface.php]: Done. In the same way I did it here: https://github.com/symfony/symfony/pull/32935/files#diff-565fb1563b9eb5b65d8a69d08ed56b2dR35

**Review by nicolas-grekas — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @vudaltsov.

**Pull request merged**